projects/igniteui-angular/src/lib/combo/combo.component.ts

```diff
@@ -419,15 +434,18 @@ export class IgxComboComponent extends IgxComboBaseDirective implements AfterVie
         }
     }
 
-    protected setSelection(selection: Set<any>, event?: Event): void {
+    protected setSelection(selection: Set<any>, event?: Event, emitEvent: boolean = true): void {
+        const filteredSelection = this.isRemote ? Array.from(selection) : Array.from(selection).filter(item => this.isItemInData(item));
+
         const currentSelection = this.selectionService.get(this.id);
-        const removed = this.convertKeysToItems(diffInSets(currentSelection, selection));
-        const added = this.convertKeysToItems(diffInSets(selection, currentSelection));
-        const newValue = Array.from(selection);
+        const removed = this.convertKeysToItems(diffInSets(currentSelection, new Set(filteredSelection)));
+        const added = this.convertKeysToItems(diffInSets(new Set(filteredSelection), currentSelection));
+        const newValue = filteredSelection;
```

Comment:
`filteredSelection` is the new value now so we can move this to be at the top of the method.